packages/salesforcedx-sobjects-faux-generator/src/describe/sObjectDescribe.ts

```diff
@@ -161,30 +162,76 @@ export enum SObjectCategory {
   CUSTOM = 'CUSTOM'
 }
 
+type SubRequest = { method: string; url: string };
+type BatchRequest = { batchRequests: SubRequest[] };
+
+type SubResponse = { statusCode: number; result: SObject };
+
+type BatchResponse = { hasErrors: boolean; results: SubResponse[] };
+
 export class SObjectDescribe {
+  private accessToken: string;
+  private instanceUrl: string;
+  // TODO should get the proper version from ??
+  private readonly servicesPath: string = 'services/data';
+  private readonly targetVersion = '40.0';
+  private readonly versionPrefix = 'v' + this.targetVersion;
+  private readonly sobjectsPart: string = this.versionPrefix + '/sobjects';
+  private readonly batchPart: string = this.versionPrefix + '/composite/batch';
+
+  // get the token and url by calling the org - short term, should be able to get it from the sfdx project
+  private async setupConnection(projectPath: string, username?: string) {
+    if (!this.accessToken) {
+      let orgInfo: any;
+      const builder = new SfdxCommandBuilder().withArg('force:org:display');
+      if (username) {
+        builder.args.push('--targetusername', username);
+      }
+      const command = builder.withJson().build();
+      const execution = new CliCommandExecutor(command, {
+        cwd: projectPath
+      }).execute();
+      const cmdOutput = new CommandOutput();
+      const result = await cmdOutput.getCmdResult(execution);
+      orgInfo = JSON.parse(result).result;
+      this.accessToken = orgInfo.accessToken;
+      this.instanceUrl = orgInfo.instanceUrl;
+    }
+  }
   public async describeSObject(
     projectPath: string,
     type: string,
     username?: string
   ): Promise<SObject> {
-    const builder = new SfdxCommandBuilder()
-      .withArg('force:schema:sobject:describe')
-      .withFlag('--sobjecttype', type);
-    if (username) {
-      builder.args.push('--targetusername', username);
-    }
-    const command = builder.withJson().build();
-    const execution = new CliCommandExecutor(command, {
-      cwd: projectPath
-    }).execute();
+    await this.setupConnection(projectPath);
+
+    const urlElements = [
+      this.instanceUrl,
+      this.servicesPath,
+      this.sobjectsPart,
+      type,
+      'describe'
+    ];
+
+    const requestUrl = urlElements.join('/');
+
+    const options: XHROptions = {
+      type: 'GET',
+      url: requestUrl,
+      headers: {
+        'Content-Type': 'application/json',
+        Accept: 'application/json',
```

Comment:
'Content-Type' in quotes but not 'Accept'? Probably should try to be consistent throughout unless you have a particular reason for that.

Reply:
just a mistake - will fix

Reply:
Hmm, you are probably right.  I just copied this from BaseCommand.ts that Jonathan pointed me at - and that I was going to unify later on how we do REST calls....

Reply:
I know why this happens. It's because of the prettier formatter removing unnecessary quotes.